### PR TITLE
Upgrade PHPUnit to 10.1.x

### PR DESCRIPTION
**Pull request by gsteel**
Closes #196

### CODE REVIEW

File: test/Helper/Navigation/AbstractTestCase.php [hunk 1]
```diff
@@ -175,7 +179,7 @@ protected function getAcl(): array
 
         $acl->addResource(new GenericResource('guest_foo'));
         $acl->addResource(new GenericResource('member_foo'), 'guest_foo');
-        $acl->addResource(new GenericResource('admin_foo', 'member_foo'));
+        $acl->addResource(new GenericResource('admin_foo'));
```
Comment (froschdesign) [inline test/Helper/Navigation/AbstractTestCase.php:182]: It looks like it doesn't matter, but I think that was the original intention:
```suggestion
        $acl->addResource(new GenericResource('admin_foo'), 'member_foo');
```

Comment (gsteel) [reply test/Helper/Navigation/AbstractTestCase.php:182]: Yes it was, but the string fixtures being compared against fail when `admin_foo` is given a parent role, so I just deleted the extra argument

Comment (froschdesign) [reply test/Helper/Navigation/AbstractTestCase.php:182]: I mean the difference between the brackets that leads to different arguments:

```php
$acl->addResource(new GenericResource('admin_foo', 'member_foo'));
// vs.
$acl->addResource(new GenericResource('admin_foo'), 'member_foo');
```

Comment (froschdesign) [reply test/Helper/Navigation/AbstractTestCase.php:182]: See also: https://github.com/laminas/laminas-view/pull/207/files#diff-3208225af5a3ef0c924761633ad6b04cbfa5eab3d3bafcc49c79441e89257ee7L2692-L2694

Comment (gsteel) [reply test/Helper/Navigation/AbstractTestCase.php:182]: Acknowledged. I am aware of the original mistake. Removing the offending and redundant parameter affects nothing, but correcting the mistake yields different navigation structures due to the change to the ACL configuration, which in turn causes tests to fail needlessly. i.e. it's not worth correcting the misplaced parenthesis.

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Comment by Ocramius** [comment]
Thanks @gsteel!

I don't think a release is needed here though, as nothing user-facing was merged.

**Pull request merged**